app/assets/javascripts/app/views/publisher/uploader_view.js

```diff
@@ -5,32 +5,46 @@
 // progress. Attaches previews of finished uploads to the publisher.
 
 app.views.PublisherUploader = Backbone.View.extend({
-
   allowedExtensions: ["jpg", "jpeg", "png", "gif", "tif", "tiff"],
   sizeLimit: 4194304,  // bytes
 
   initialize: function(opts) {
     this.publisher = opts.publisher;
-
-    this.uploader = new qq.FileUploaderBasic({
+    this.uploader = new qq.FineUploaderBasic({
       element: this.el,
-      button:  this.el,
+      button: this.el,
 
       //debug: true,
 
-      action: "/photos",
-      params: { photo: { pending: true }},
-      allowedExtensions: this.allowedExtensions,
-      sizeLimit: this.sizeLimit,
+      request: {
+        endpoint: "/photos",
+        params: {
+          /* eslint-disable camelcase */
+          authenticity_token: $("meta[name='csrf-token']").attr("content"),
+          /* eslint-enable camelcase */
+          photo: {
+            pending: true
+          }
+        },
+        paramsInBody: false
+      },
+      validation: {
+        allowedExtensions: this.allowedExtensions,
+        sizeLimit: this.sizeLimit
+      },
       messages: {
-        typeError:  Diaspora.I18n.t("photo_uploader.invalid_ext"),
-        sizeError:  Diaspora.I18n.t("photo_uploader.size_error"),
+        typeError: Diaspora.I18n.t("photo_uploader.invalid_ext"),
+        sizeError: Diaspora.I18n.t("photo_uploader.size_error"),
         emptyError: Diaspora.I18n.t("photo_uploader.empty")
       },
-      onProgress: _.bind(this.progressHandler, this),
-      onSubmit:   _.bind(this.submitHandler, this),
-      onComplete: _.bind(this.uploadCompleteHandler, this)
-
+      callbacks: {
+        onProgress: _.bind(this.progressHandler, this),
+        onSubmit: _.bind(this.submitHandler, this),
+        onComplete: _.bind(this.uploadCompleteHandler, this),
+        onError: function(id, name, errorReason) {
+          alert(errorReason);
```

Comment:
When using FineUploader, the `onError` callback is triggered when an invalid file type or a file with a size above limit has been selected to upload. This is a difference compared to the custom file uploader we used before.  

After removing the callback, the user won't see any message anymore when trying to upload a 5 MB picture.
Indeed in very rare cases users could also see a confusing message like the one you posted, but in most cases
![screenshot_20170107_160739](https://cloud.githubusercontent.com/assets/1684826/21742704/d0da2c54-d4f3-11e6-81b5-373e983fb836.png)
 the alert (yeah, I prefer flash messages too) makes sense.